rclcpp/include/rclcpp/node_handle.hpp

```diff
@@ -0,0 +1,280 @@
+// Copyright 2022 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP__NODE_HANDLE_HPP_
+#define RCLCPP__NODE_HANDLE_HPP_
+
+#include <memory>
+#include <type_traits>
+
+#include "rclcpp/node_interfaces/node_base_interface.hpp"
+#include "rclcpp/node_interfaces/node_clock_interface.hpp"
+#include "rclcpp/node_interfaces/node_graph_interface.hpp"
+#include "rclcpp/node_interfaces/node_logging_interface.hpp"
+#include "rclcpp/node_interfaces/node_parameters_interface.hpp"
+#include "rclcpp/node_interfaces/node_services_interface.hpp"
+#include "rclcpp/node_interfaces/node_time_source_interface.hpp"
+#include "rclcpp/node_interfaces/node_timers_interface.hpp"
+#include "rclcpp/node_interfaces/node_topics_interface.hpp"
+#include "rclcpp/node_interfaces/node_waitables_interface.hpp"
+
+namespace rclcpp
+{
+
+using BaseInterface = node_interfaces::NodeBaseInterface::SharedPtr;
+using ClockInterface = node_interfaces::NodeClockInterface::SharedPtr;
+using GraphInterface = node_interfaces::NodeGraphInterface::SharedPtr;
+using LoggingInterface = node_interfaces::NodeLoggingInterface::SharedPtr;
+using TimersInterface = node_interfaces::NodeTimersInterface::SharedPtr;
+using TopicsInterface = node_interfaces::NodeTopicsInterface::SharedPtr;
+using ServicesInterface = node_interfaces::NodeServicesInterface::SharedPtr;
+using WaitablesInterface = node_interfaces::NodeWaitablesInterface::SharedPtr;
+using ParametersInterface = node_interfaces::NodeParametersInterface::SharedPtr;
+using TimeSourceInterface = node_interfaces::NodeTimeSourceInterface::SharedPtr;
```

Comment:
I think these renames are misleading because they don't imply that a `shared_ptr` is being used.

Actually, my preference would be to not have these aliases at all. In header files I prefer (and it's the prevailing style of `rclcpp`) to use full names and not create unnecessary aliases. In `_impl.hpp` files and `.cpp` files you can make any usings you want, but we should avoid them in headers, imo.

Also, if you do use them in headers, ideally do so in the scope of a function or if on a class, make them private.

Reply:
And if the intention was to make shorthand for the users of this class, then I would recommend against that. The users can make shortcuts if they want.